tests/test_tiatoolbox.py

```diff
@@ -454,3 +452,60 @@ def test_command_line_save_tiles(_response_all_wsis, tmp_path):
     )
 
     assert save_tiles_result.exit_code == 0
+
+
+def test_wsimeta_init_fail():
+    with pytest.raises(TypeError):
+        wsimeta.WSIMeta(slide_dimensions=None)
+
+
+def test_wsimeta_validate_fail():
+    meta = wsimeta.WSIMeta(slide_dimensions=(512, 512), level_dimensions=[])
+    assert meta.validate() is False
+
+    meta = wsimeta.WSIMeta(
+        slide_dimensions=(512, 512),
+        level_dimensions=[(512, 512), (256, 256)],
+        level_count=3,
+    )
+    assert meta.validate() is False
+
+    meta = wsimeta.WSIMeta(slide_dimensions=(512, 512), level_downsamples=[1, 2],)
+    assert meta.validate() is False
+
+    meta = wsimeta.WSIMeta(slide_dimensions=(512, 512), level_downsamples=[1, 2],)
+    assert meta.validate() is False
+
+    meta = wsimeta.WSIMeta(slide_dimensions=(512, 512))
+    meta.level_dimensions = None
+    assert meta.validate() is False
+
+    meta = wsimeta.WSIMeta(slide_dimensions=(512, 512))
+    meta.level_downsamples = None
+    assert meta.validate() is False
+
+
+def test_wsimeta_validate_pass():
+    meta = wsimeta.WSIMeta(slide_dimensions=(512, 512))
+    assert meta.validate()
+
+    meta = wsimeta.WSIMeta(
+        slide_dimensions=(512, 512),
+        level_dimensions=[(512, 512), (256, 256)],
+        level_downsamples=[1, 2],
+    )
+    assert meta.validate()
+
+
+def test_wsimeta_openslidewsireader_ndpi(_response_ndpi, tmp_path):
+    input_dir, file_name, ext = utils.misc.split_path_name_ext(str(_response_ndpi))
+    wsi_obj = wsireader.OpenSlideWSIReader(input_dir, file_name + ext)
+    meta = wsi_obj.slide_info
+    assert meta.validate()
+
+
+def test_wsimeta_openslidewsireader_svs(_response_svs, tmp_path):
+    input_dir, file_name, ext = utils.misc.split_path_name_ext(str(_response_svs))
+    wsi_obj = wsireader.OpenSlideWSIReader(input_dir, file_name + ext)
```

Comment:
Just to be clear, are we keeping the input as before or changing it to full file path?